pkg/deploy/generator/resources.go

```diff
@@ -113,14 +113,14 @@ func (g *generator) virtualNetwork(name, addressPrefix string, subnets *[]mgmtne
 
 // virtualNetworkPeering configures vnetA to peer with vnetB, two symmetrical
 // configurations have to be applied for a peering to work
-func (g *generator) virtualNetworkPeering(name, vnetB string) *arm.Resource {
+func (g *generator) virtualNetworkPeering(name, vnetB string, allowGatewayTransit, useRemoteGateways bool, dependsOn []string) *arm.Resource {
```

Comment:
[Gateway transit](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-peering-overview#gateways-and-on-premises-connectivity) is a peering property that lets one virtual network use the VPN gateway in the peered virtual network for cross-premises or VNet-to-VNet connectivity

We want to be able to Allow Gateway Transit / Use Remote Gateways for the particular peerings: 

```
dev-vpn-vnet -> dev-vnet [Allow Gateway Transit = True, Use Remote Gateways = False]
dev-vnet -> dev-vpn-vnet [Allow Gateway Transit = False, Use Remote Gateways = True]
dev-vpn-vnet -> rp-vnet [Allow Gateway Transit = True, Use Remote Gateways = False]
rp-vnet -> dev-vpn-vnet [Allow Gateway Transit = False, Use Remote Gateways = True]
```

Reply:
@karanmagdani somehow the dev-vnet still had a /9 in the generator but all the templates were correct. I changed the /9 to a /16 to match what we had talked about.

Reply:
Thanks @zacharyljones , i think i may have missed that as part of my PR against your branch. Also feel free to add any comments  that you believe will help reviewers